### PR TITLE
[MIG] sale_discount_display_amount: Migration to 17.0

### PR DESCRIPTION
**Pull request by FSchiepers**
This version adds the possibility to choose whether the discount is calculated based on the non-taxed subtotal or the taxed total, see: https://github.com/OCA/sale-workflow/issues/2813

Correction from https://github.com/OCA/sale-workflow/pull/2987

### CODE REVIEW

File: sale_discount_display_amount/__manifest__.py [hunk 1]
```diff
@@ -6,12 +6,15 @@
...
         sale_order_line and sale_order level""",
-    "version": "16.0.1.0.0",
+    "version": "17.0.1.0.0",
     "license": "AGPL-3",
     "author": "ACSONE SA/NV,Odoo Community Association (OCA)",
     "website": "https://github.com/OCA/sale-workflow",
     "depends": ["sale_management"],
-    "data": ["views/sale_view.xml"],
+    "data": ["views/sale_view.xml", "views/res_config_settings_views.xml"],
+    "installable": True,
```
Comment (rousseldenis) [inline sale_discount_display_amount/__manifest__.py]: @FSchiepers All these values are the default ones, not necessary

File: sale_discount_display_amount/hooks.py [hunk 2]
```diff
@@ -2,41 +2,48 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 import logging
 
-from odoo import SUPERUSER_ID
-from odoo.api import Environment
 from odoo.tools.sql import column_exists, create_column
 
-_logger = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
```
Comment (rousseldenis) [inline sale_discount_display_amount/hooks.py]: This notation is widely used to indicate that should not be used externally. You should let it as is.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review.

IMHO changes in the module should be done in a separate commit in order to separate the real migration and the improvements.

**Comment by rousseldenis** [comment]
@FSchiepers Could you improve your commit messages :

image


Something like `[IMP] sale_discount_display_amount: (...)`

Thanks

File: sale_discount_display_amount/hooks.py
Comment (FSchiepers) [inline sale_discount_display_amount/hooks.py:1]: [IMP]sale_discount_display_amount: : renamed Logger to _logger

File: sale_discount_display_amount/__manifest__.py
Comment (FSchiepers) [inline sale_discount_display_amount/__manifest__.py:1]: [IMP] sale_discount_display_amount: Removed all the standard parameters.

**Review by FSchiepers — COMMENTED** [review]
[IMP]sale_discount_display_amount: : renamed Logger to _logger

**Review by FSchiepers — COMMENTED** [review]
[IMP] sale_discount_display_amount: Removed all the standard parameters.

**Comment by FSchiepers** [comment]
I added comments, can you review please?

**Comment by FSchiepers** [comment]
rebased to https://github.com/OCA/sale-workflow/pull/3160

**Pull request closed without merging**